arangod/Metrics/ClusterMetricsFeature.cpp

```diff
@@ -67,130 +67,248 @@ void ClusterMetricsFeature::validateOptions(
   }
 }
 
-void ClusterMetricsFeature::start() { asyncUpdate(); }
+void ClusterMetricsFeature::start() {
+  if (wasStop()) {
+    return;
+  }
+  auto& ci = server().getFeature<ClusterFeature>().clusterInfo();
+  ci.initMetricsState();
+  if (_timeout != 0) {
+    rescheduleTimer();
+  }
+}
 
 void ClusterMetricsFeature::beginShutdown() {
   _count.store(kStop, std::memory_order_release);
-  _handle = nullptr;
+  std::atomic_store_explicit(&_timer, {}, std::memory_order_relaxed);
+  std::atomic_store_explicit(&_update, {}, std::memory_order_relaxed);
 }
 
-void ClusterMetricsFeature::asyncUpdate() {
-  if (_count.load(std::memory_order_acquire) % 2 != kStop &&
-      _count.fetch_add(kUpdate, std::memory_order_acq_rel) == 0) {
-    scheduleUpdate();
+void ClusterMetricsFeature::stop() { beginShutdown(); }
+
+std::optional<std::string> ClusterMetricsFeature::update(
+    CollectMode mode) noexcept {
+  if (mode == CollectMode::TriggerGlobal) {
+    auto const count = _count.fetch_add(kUpdate, std::memory_order_acq_rel);
+    if (count == 0) {
+      rescheduleUpdate(0);
+    }
+    return std::nullopt;
   }
+  TRI_ASSERT(mode != CollectMode::Local);
+  auto& nf = server().getFeature<NetworkFeature>();
+  auto& cf = server().getFeature<ClusterFeature>();
+  auto& ci = cf.clusterInfo();
+  try {
+    // TODO(MBkkt) Need to return Future<std::optional<std::string>>:
```

Comment:
This comment about I can replace some blocking waiting with non-blocking (automaton in RestHandler) 
For make it I needed to understand some un/bad-documented part of agency API.
So I decided to delay it, because it's not critical, and I just wrote a comment about that is possible.